### PR TITLE
Add support for the google_compute_service_attachment resource

### PR DESCRIPTION
**Pull request by rileykarson**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/9062

~Using `ACCEPT_MANUAL` and project(s) set causes a permadiff, I'm awaiting a response about that before moving forward~

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_compute_service_attachment`
```

### CODE REVIEW

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -14119,6 +14119,135 @@ objects:
...
+    kind: 'compute#ServiceAttachment'
+    base_url: projects/{{project}}/regions/{{region}}/serviceAttachments
+    has_self_link: true
+    input: true
+    min_version: beta
+    description: |
+      Represents a ServiceAttachment resource.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'TODO': 'https://cloud.google.com/'
```
Comment (slevenick) [inline mmv1/products/compute/api.yaml:14133]: Do we have a guide to include here now?

File: mmv1/products/compute/terraform.yaml [hunk 2]
```diff
@@ -2198,6 +2198,38 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           If it is not provided, the provider region is used.
   SecurityPolicy: !ruby/object:Overrides::Terraform::ResourceOverride
     exclude: true
+  ServiceAttachment: !ruby/object:Overrides::Terraform::ResourceOverride
+    examples:
```
Comment (slevenick) [inline mmv1/products/compute/terraform.yaml:2202]: Do these need min_version beta?

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (rileykarson) [reply mmv1/products/compute/api.yaml]: Yep, added. Good catch!

**Review by slevenick — APPROVED** [review]
Seems reasonable to me once `ACCEPT_MANUAL` works

**Comment by rileykarson** [comment]
Oh! That does now, that was [`1f28c23` (#4855)](https://github.com/GoogleCloudPlatform/magic-modules/pull/4855/commits/1f28c23db569fe5c3de339c007aed4c1f148bd3f)

**Pull request merged**